lib/3rd_party_adapters/LVGL/Screen.cpp

```diff
@@ -0,0 +1,472 @@
+
+    auto item = reinterpret_cast<const MenuItemSubmenu *const>(lv_event_get_user_data(e));
+
+    if ((code == LV_EVENT_SHORT_CLICKED) && (item != nullptr))
+    {
+        //if we were clicked shortly, draw the next menu screen
+        lv_obj_clean(lv_scr_act());
+        auto ptrSubMenuScreen = new ScreenMenu{*item->getSubMenuList()};
+        ptrSubMenuScreen->draw();
+    }
+    else if (code == LV_EVENT_KEY)
+    {
+        uint32_t key = lv_event_get_key(e);
+        if (key == LV_KEY_ESC)
+        {
+            //if we receive the back button, go one screen back
+            IScreen_leave();
+        }
+    }
+}
+
+/**
+ * @brief Event callback function for a switch item that modifies a bool variable
+ * @note  The lvgl event user data hold a pointer to the triggered switch item
+ * 
+ * @param e - pointer to lvgl event object
+ */
+static void ScreenMenu_switch_cb(lv_event_t *e)
+{
+    lv_event_code_t code = lv_event_get_code(e);
+    lv_obj_t *obj = lv_event_get_target(e);
+
+    auto item = reinterpret_cast<const MenuItemSwitch *const>(lv_event_get_user_data(e));
+
+    if ((code == LV_EVENT_VALUE_CHANGED) && (item != nullptr))
+    {
+        //if we were clicked the switch, change the value of the original variable
+        bool *ptr = item->getPtrBool();
+        if (ptr)
+        {
+            *ptr = lv_obj_has_state(obj, LV_STATE_CHECKED) ? 1 : 0;
+        }
+    }
+    else if (code == LV_EVENT_KEY)
+    {
+        uint32_t key = lv_event_get_key(e);
+        if (key == LV_KEY_ESC)
+        {
+            //if we receive the back button, go one screen back
+            IScreen_leave();
+        }
+    }
+}
+
+/**
+ * @brief Event callback function for a value item that modifies a float variable
+ * @note  The lvgl event user data hold a pointer to the triggered value item
+ * 
+ * @param e - pointer to lvgl event object
+ */
+static void ScreenMenu_value_cb(lv_event_t *e)
+{
+    lv_event_code_t code = lv_event_get_code(e);
+    lv_obj_t *obj = lv_event_get_target(e);
+
+    auto item = reinterpret_cast<const MenuItemValue *const>(lv_event_get_user_data(e));
+
+    if ((code == LV_EVENT_SHORT_CLICKED) && (item != nullptr))
+    {
+        //if we were clicked shortly, draw the value modification screen
+        auto ValModifyScreen = new ScreenValueModifier{item};
+        ValModifyScreen->draw();
+    }
+    else if (code == LV_EVENT_KEY)
+    {
+        uint32_t key = lv_event_get_key(e);
+        if (key == LV_KEY_ESC)
+        {
+            //if we receive the back button, go one screen back
+            IScreen_leave();
+        }
+    }
+}
+
+/**
+ * @brief Construct a new ScreenMenu object
+ * 
+ * @param itemList - List of items to be drawn with this menu
+ */
+ScreenMenu::ScreenMenu(MenuItemList itemList)
+    : _List{itemList}
+{
+}
+
+/**
+ * @brief Draw function of the ScreenMenu object
+ *  
+ *   This function translates the list of item types into actual lvgl draw directives.
```

Comment:
This brief description has no value.

```suggestion
 * @brief Translates the list of item types into actual LVGL draw directives.
```

Reply:
The descriptions have been shortened.